File: tests/07_phases_ompss.dg/cxx/success_task_reduction_01.cpp
/*
<testinfo>
test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
</testinfo>
*/

namespace N
{
    struct A
    {
        void foo();

        void bar()
        {
            int x = 0;

            #pragma omp task reduction(+: x)
            {
                x++;
            }

            #pragma omp taskwait
        }
    };

    void A::foo()
    {
        int x = 0;

        #pragma omp task reduction(+: x)
        {
            x++;
        }

        #pragma omp taskwait
    }
}

int main()
{
    N::A a;
    a.foo();
    a.bar();
}

File: tests/07_phases_ompss_2.dg/fortran/success_task_reduction_01.f90
! <testinfo>
! test_generator=config/mercurium-ompss-2
! </testinfo>


! Reductions scalar

PROGRAM P
    INTEGER(4) :: X
    X = 0
    DO I=1,1000
        !$OSS TASK REDUCTION(+: X)
            X = X + 1;
        !$OSS END TASK
    END DO
    !$OSS TASKWAIT

    IF (X /= 1000) STOP 1

END PROGRAM P

